File: event-bus-client/src/main/java/pegasus/eventbus/client/EventResult.java
package pegasus.eventbus.client;

/**
 * An enumeration that defines the values that may be returned from the {@link EventHandler#handleEvent} 
 * 		and {FallbackHandler#HandleEnvelope} methods.
 */
public enum EventResult {
	/**
	 * Indicates that the event was successfully handled.
	 */
	Handled,
	
	/**
	 * Indicates that the event could not be handled and should not be retried. 
	 */
	Failed,
	
	/**
	 * Indicates that the event could not be handled at this time.  The event will be placed at 
	 * 		the back of the queue and re-raised when it reaches the front again.
	 */
	Retry
}

File: event-bus-client/src/main/dotnet/EventResult.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace pegasus.eventbus.client
{
    /// <summary>
    /// An enumeration that defines the values that may be returned from the <see cref="IEventHandler.HandleEvent(object)"/>
    /// and <see cref="IEnvelopeHandler.HandleEnvelope(Envelope)"/> methods
    /// </summary>
    public enum EventResult
    {
        /// <summary>
        /// Indicates that the event was successfully handled.
        /// </summary>
        Handled,

        /// <summary>
        /// Indicates that the event could not be handled and should not be retried.
        /// </summary>
        Failed,

        /// <summary>
        /// Indicates that the event could not be handled at this time.  The event will be placed at 
        /// the back of the queue and re-raised when it reaches the front again.
        /// </summary>
        Retry
    }
}
